apps/cms.c

```diff
@@ -676,22 +692,19 @@ int cms_main(int argc, char **argv)
                 goto end;
             break;
         case OPT_3DES_WRAP:
-# ifndef OPENSSL_NO_DES
-            wrap_cipher = (EVP_CIPHER *)EVP_des_ede3_wrap();
-# endif
+            wrapname = first(PROV_NAMES_DES3_WRAP);
             break;
         case OPT_AES128_WRAP:
-            wrap_cipher = (EVP_CIPHER *)EVP_aes_128_wrap();
+            wrapname = first(PROV_NAMES_AES_128_WRAP);
             break;
         case OPT_AES192_WRAP:
-            wrap_cipher = (EVP_CIPHER *)EVP_aes_192_wrap();
+            wrapname = first(PROV_NAMES_AES_128_WRAP);
             break;
         case OPT_AES256_WRAP:
-            wrap_cipher = (EVP_CIPHER *)EVP_aes_256_wrap();
+            wrapname = first(PROV_NAMES_AES_128_WRAP);
             break;
```

Comment:
I would actually reduce all the specific wrap cases like this:

``` C
        case OPT_3DES_WRAP:
        case OPT_AES128_WRAP:
        case OPT_AES192_WRAP:
        case OPT_AES256_WRAP:
            wrapname = opt_flag() + 1;
            break;
```

The option strings happen to be perfectly legitimate names already, so...
With this, no need for first(), no need to move prov/names.h, ...

Reply:
That's clever!  Does fetch do strcasecmp or strcmp?
```
#define PROV_NAMES_AES_128_WRAP "AES-128-WRAP:id-aes128-wrap:AES128-WRAP:2.16.840.1.101.3.4.1.5"
```

Reply:
strcasecmp

Reply:
> strcasecmp

Okay, that's great.  I'll do what you suggested.  I will also change `test/recipes/30-test_evp_data/evpciph_aes_wrap.txt` to have an uppercase and lowercase name.